mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_FileManager.java

```diff
@@ -90,38 +90,34 @@ public static void copyStream(final InputStream inStream, final OutputStream out
         }
     }
 
-    public static HashMap<String, String> getScriptInfo(final String js) {
-        final HashMap<String, String> map = new HashMap<String, String>();
+    public static PluginInfo getScriptInfo(final String js) {
+        final PluginInfo info = new PluginInfo();
         String header = "";
         // get metadata of javascript file
         if (js != null && js.contains("==UserScript==") && js.contains("==/UserScript==")) {
             header = js.substring(js.indexOf("==UserScript=="),
                     js.indexOf("==/UserScript=="));
         }
         // add default values
-        map.put("id", "unknown");
-        map.put("version", "not found");
-        map.put("name", "unknown");
-        map.put("description", "");
-        map.put("category", "Misc");
+        info.setId("unknown");
+        info.setVersion("not found");
+        info.setName("unknown");
+        info.setDescription("");
+        info.setCategory("Misc");
```

Comment:
It looks like this is the only place the explicit setters are used, I feel this can be set in the PluginInfo constructor